src/ui/api/email.go

```diff
@@ -32,20 +30,20 @@ const (
 
 // EmailAPI ...
 type EmailAPI struct {
-	api.BaseAPI
+	BaseController
 }
 
 // Prepare ...
 func (e *EmailAPI) Prepare() {
-	userID := e.ValidateUser()
-	isSysAdmin, err := dao.IsAdminRole(userID)
-	if err != nil {
-		log.Errorf("failed to check the role of user: %v", err)
-		e.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+	e.BaseController.Prepare()
```

Comment:
Yes, we can add a filter to check the authentication. But it's complicate as not all API calls need authentication, we need a filter to do the authentication according to the url, method, etc. Maybe we can refactor this part after completing the current work.